app/actions/create-memory.ts

```diff
@@ -0,0 +1,45 @@
+"use server";
+
+import axios from "axios";
+import { Memory } from "../models/memory.model";
+import { revalidatePath } from "next/cache";
+import { MemoryDTO } from "@/app/dto/memory-dto";
+
+type File = {
+  type: string;
+  name: string;
+};
+
+function isImageFile(file: any): file is File {
+  return file && file?.type?.startsWith("image/");
+}
+
+async function createImageUrl(image: FormDataEntryValue): Promise<string> {
+  return isImageFile(image)
+    ? `data:${image.type};base64,${Buffer.from(
+        await image.arrayBuffer()
+      ).toString("base64")}`
+    : "";
+}
```

Comment:
it's important to note (and I forgot to mention it in the README) that I would use a service like Cloudinary or any other image management service to handle this image, instead of transforming it in a base 64 in a real project